### PR TITLE
Adding test for prep-file generation

### PR DESCRIPTION
**Pull request by charles-cowart**
Test against proper testing environment.

### CODE REVIEW

File: qp_klp/tests/test_klp.py [hunk 1]
```diff
@@ -395,6 +407,30 @@ def test_sequence_processing_pipeline(self):
 
         self.assertEqual(ainfo, exp)
 
+        # confirm that 'cmds.log' exists.
+        cmd_log_fp = join(self.out_dir, 'cmds.log')
+        self.assertTrue(exists(cmd_log_fp))
+
+        # confirm that fastq files were copied to uploads directory.
+        uploads_fp = ('/home/runner/work/qp-knight-lab-processing/'
```
Comment (antgonza) [inline qp_klp/tests/test_klp.py:415]: you need to take this path from the configuration, right? If not it's gonna fail in other systems.

**Review by antgonza — CHANGES_REQUESTED** [review]
Looking good, some minor comments. Do you think is worth checking the contents of the prep info file? Thank you!

File: qp_klp/tests/test_klp.py [hunk 2]
```diff
@@ -307,7 +307,19 @@ def test_sequence_processing_pipeline(self):
...
         qcj_filtered_sequences = join(qcj_output_fp, 'filtered_sequences')
         makedirs(qcj_filtered_sequences)
-        makedirs(join(qcj_output_fp, 'fastp_reports_dir', 'json'))
+
+        # create GenPrepFileJob output directory for use by packaging code and
+        # downstream-testing.
+        gp_root_fp = join(self.out_dir, 'GenPrepFileJob')
+        prep_files_root_fp = join(gp_root_fp, 'PrepFiles')
+        makedirs(prep_files_root_fp, exist_ok=True)
+        # TODO: Fix name
```
[the remark this answers is not in the document]
Comment (charles-cowart) [reply qp_klp/tests/test_klp.py:316]: I'll rm. For right now it's fine IMOO.

File: qp_klp/tests/test_klp.py [hunk 3]
```diff
@@ -395,6 +407,30 @@ def test_sequence_processing_pipeline(self):
...
+        self.assertTrue(exists(cmd_log_fp))
+
+        # confirm that fastq files were copied to uploads directory.
+        uploads_fp = ('/home/runner/work/qp-knight-lab-processing/'
+                      'qp-knight-lab-processing/qiita-dev/qiita_db/'
+                      'support_files/test_data/uploads/11661')
+
+        for some_file in new_files:
+            some_path = join(uploads_fp, some_file)
+            print("checking '%s' exists..." % some_path)
```
[the remark this answers is not in the document]
Comment (charles-cowart) [reply qp_klp/tests/test_klp.py:421]: ty!

**Comment by charles-cowart** [comment]
@antgonza It would be good to check the contents of the prep-file results, but I think it would be better to refactor the tests first now that the plugin and the pipeline are mature. Most of the functionality is currently tested in one large test and it requires the full qiita dependency. Some of the functionality and file/directory structures are mocked-up for testing and those mock-ups aren't high-fidelity.

**Comment by antgonza** [comment]
Thank you @charles-cowart !

**Review by antgonza — APPROVED** [review]

**Pull request merged**